docs/sdks/typescript/quickstart.md

```diff
@@ -142,365 +149,513 @@ function App() {
+code {
+  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
+    monospace;
+}
 
-Each user has a set of credentials, which consists of two parts:
+/* ----- Buttons ----- */
+button {
+  padding: 0.5rem 0.75rem;
+  border: none;
+  border-radius: 0.375rem;
+  background-color: var(--theme-color);
+  color: var(--theme-color-contrast);
+  cursor: pointer;
+  font-weight: 600;
+  letter-spacing: 0.1px;
+  font-family: monospace;
+}
 
-- An `Identity`, a unique public identifier. We're using these to identify `User` rows.
-- A `Token`, a private key which SpacetimeDB uses to authenticate the client.
+/* ----- Inputs & Textareas ----- */
+input,
+textarea {
+  border: none;
+  border-radius: 0.375rem;
+  caret-color: var(--theme-color);
+  font-family: monospace;
+  font-weight: 600;
+  letter-spacing: 0.1px;
+  padding: 0.5rem 0.75rem;
+}
 
-These credentials are generated by SpacetimeDB each time a new client connects, and sent to the client so they can be saved, in order to re-connect with the same identity.
+input:focus,
+textarea:focus {
+  outline: none;
+  box-shadow: 0 0 0 2px var(--theme-color);
+}
+```
 
-We want to store our local client identity in a stateful variable and also save our `token` to local storage for future connections.
+Now when you run `pnpm run dev` and open `http://localhost:5173`, you should see a basic chat app that does not yet send or receive messages.
 
-Each client also has an `Address`, which modules can use to distinguish multiple concurrent connections by the same `Identity`. We don't need to know our `Address`, so we'll ignore that argument.
+## Generate your module types
 
-Once we are connected, we can send our subscription to the SpacetimeDB module. SpacetimeDB is set up so that each client subscribes via SQL queries to some subset of the database, and is notified about changes only to that subset. For complex apps with large databases, judicious subscriptions can save each client significant network bandwidth, memory and computation compared. For example, in [BitCraft](https://bitcraftonline.com), each player's client subscribes only to the entities in the "chunk" of the world where that player currently resides, rather than the entire game world. Our app is much simpler than BitCraft, so we'll just subscribe to the whole database.
+The `spacetime` CLI's `generate` command will generate client-side interfaces for the tables, reducers and types defined in your server module.
 
-To the body of `App`, add:
+In your `quickstart-chat` directory, run:
 
-```typescript
-client.current.onConnect((token, identity, address) => {
-  console.log('Connected to SpacetimeDB');
+```bash
+mkdir -p client/src/module_bindings
+spacetime generate --lang typescript --out-dir client/src/module_bindings --project-path server
+```
 
-  local_identity.current = identity;
+> This command assumes you've already created a server module in `quickstart-chat/server`.
```

Comment:
```suggestion
> This command assumes you've already created a server module in `quickstart-chat/server`. If you haven't, go follow either the [Rust](/docs/modules/rust/quickstart) or [C#](/docs/modules/c-sharp/quickstart) module quickstart, then come back here.
```

I know it's linked above, but we should assume many readers will skip the intro.